test/controllers/stages_controller_test.rb

```diff
@@ -221,13 +234,29 @@
         end
 
         describe 'valid attributes' do
+          [[true, "0"], [false, "1"]].each do |bool, val|
```

Comment:
if you submit as true/false this tests gets a bit simpler and the code makes more sense :D